### PR TITLE
[COMCTL32] Fix CORE-14619 (Checkbox buttons theming regression)

### PR DESCRIPTION
**Pull request by andy-123**
Fix cb_states / rb_states lookup table.

Wine is not affected by this bug.

CB_ThemedPaint (current git) looks a bit different and uses get_draw_state instead of lookup-tables cb_states / rb_states.

It seems in ROS CB_ThemedPaint was copied from theme_button.c. The order of ButtonState from theme_button.c is not equal to ButtonState in button.c, but it fits to cb_states/rb_states in CB_ThemedPaint.

**Edit**
This is a correction to [wine-3.3-sync from 04.April.2018](https://github.com/reactos/reactos/commit/b3fb8555bf9abcc04eeac31c7153d33e2f027fc0).

### CODE REVIEW

File: dll/win32/comctl32/button.c [hunk 1]
```diff
@@ -1898,15 +1898,15 @@ static void CB_ThemedPaint(HTHEME theme, const BUTTON_INFO *infoPtr, HDC hDC, Bu
 {
     static const int cb_states[3][5] =
     {
-        { CBS_UNCHECKEDNORMAL, CBS_UNCHECKEDHOT, CBS_UNCHECKEDPRESSED, CBS_UNCHECKEDDISABLED, CBS_UNCHECKEDNORMAL },
```
Comment (learn-more) [inline dll/win32/comctl32/button.c]: If this references data in another file it would be nice to have a comment with what it should be synchronized.

Comment (andy-123) [reply dll/win32/comctl32/button.c]: I think is from around [wine 3.2](https://source.winehq.org/git/wine.git/blob/refs/tags/wine-3.2:/dlls/comctl32/button.c#l1290)
In [current git] it has moved to [get_draw_state](https://source.winehq.org/git/wine.git/blob/HEAD:/dlls/comctl32/button.c#l372)

Comment (andy-123) [reply dll/win32/comctl32/button.c]: README.WINE is wrong. It was synced to wine 3.3 [see here](https://github.com/reactos/reactos/commit/b3fb8555bf9abcc04eeac31c7153d33e2f027fc0)
So this PR fixes syncing bug. I will correct REAME.WINE.

**Comment by JoachimHenze** [comment]
Thumbs up. I checked commit1 and see a definite improvement even for my theme-unskilled eye.
The orange focusing in Luna is back, see https://jira.reactos.org/browse/CORE-14619?focusedCommentId=115478&page=com.atlassian.jira.plugin.system.issuetabpanels:comment-tabpanel#comment-115478
I will merge commit1 into RCs and let Oleg Dubinskiy do the final judgement whether it's already perfect.

**Comment by HBelusca** [comment]
Committed, thanks to you all!

**Pull request merged**